static/app/types/profiling.d.ts

```diff
@@ -0,0 +1,62 @@
+namespace Profiling {
+  interface RawProfileBase {
+    endValue: number;
+    startValue: number;
+    name: string;
+    unit: string;
+    spans?: RawSpan[];
+  }
+
+  // Android traces follow this format
+  interface EventedProfile extends RawProfileBase {
+    events: ReadonlyArray<{at: number; frame: number; type: 'O' | 'C'}>;
+    type: 'evented';
+  }
+
+  // iOS traces follow this format
+  interface SampledProfile extends RawProfileBase {
```

Comment:
we usually only use `type` instead of `interface`

Reply:
@priscilawebdev 👍🏼 makes sense, but in this case (extending the base type) I would say that using interfaces is preferable, as they'll be more gracefully handled by the [compiler](https://github.com/microsoft/TypeScript-wiki/blob/main/Performance.md#preferring-interfaces-over-intersections). 

I've traced the sentry codebase to see why tsc is slow and in a lot of places, the compiler is actually struggling on really large intersection types - I'll share what I found with the team during the next TSC I think.

Reply:
oh, that's really interesting! thanks for investigating this 🙏

Looking forward to hearing your findings at the next TSC meeting

Reply:
Thank you and thanks for the review! :)